### PR TITLE
experiment: viper await async ghost state

### PR DESCRIPTION
**Pull request by crusso**
- adds await async await ghost variables and internal assertions 
- experiment with lighter weight helper for constructing internal nodes.

### CODE REVIEW

File: src/viper/trans.ml [hunk 1]
```diff
@@ -204,14 +226,35 @@ and stmt ctxt (s : M.exp) : seqn =
...
+     in
+
+     ctxt.ghost_inits := mk_s :: !(ctxt.ghost_inits);
+     { !!([],
+          [ !!(FieldAssignS(
+              (self ctxt Source.no_region, id),
+              (!!(AddE(!!(FldAcc (self ctxt Source.no_region, id)),
+                       !!(IntLitE (Mo_values.Numerics.Int.of_int 1)))))));
+            !!(ExhaleS (!!(BoolLitE true)));
+            !!(InhaleS (!!(BoolLitE true)));
```
Comment (crusso) [inline src/viper/trans.ml:250]: Complete me...

File: src/viper/trans.ml [hunk 2]
```diff
@@ -5,6 +5,23 @@ open Syntax
 module T = Mo_types.Type
 module M = Mo_def.Syntax
 
+module Stamps = Env.Make(String)
+
+let stamps : int Stamps.t ref = ref Stamps.empty
+
+let (!!) it = { it; at = Source.no_region; note = NoInfo}
```
Comment (ggreif) [inline src/viper/trans.ml:12]: thank you :-)

File: src/viper/trans.ml (hunk 1 above)
Comment (ggreif) [reply src/viper/trans.ml:250]: Both should be `MacroCall`

File: src/viper/trans.ml (hunk 2 above)
Comment (aterga) [reply src/viper/trans.ml:12]:
```suggestion
let (!!) it = { it; at = Source.no_region; note = NoInfo }
```

;-)

Comment (aterga) [reply src/viper/trans.ml:12]: Btw, what's your intuition for calling this symbol `!!`?

**Comment by aterga** [comment]
This is great, thank you @crusso ! I spotted only one thing that's missing: after the first havoc, we must inhale `self._message_async > 0`, meaning that the code block after the havoc cannot be executed unless this block is actually entered. Does this make sense? See Martin's snippet here: https://dfinity.slack.com/archives/C03AWN58T32/p1666097506313909

File: src/viper/trans.ml (hunk 2 above)
Comment (crusso) [reply src/viper/trans.ml]: None.

**Comment by crusso** [comment]
> 

Good catch. Will fix then merge

**Pull request merged**